### ISSUE TITLE
WPML installation stopped working?

### ISSUE THREAD

**Issue by Twansparant** [ffraenz/private-composer-installer#41]
Hi there,

As of today, we're no longer able to update WPML plugins through the installer?
Locally everything works fine, but on remote servers `composer install` fails with these errors:

```
[Composer\\Downloader\\TransportException]
The 'https://wpml.org/?download=8882&user_id={my_user_id}&subscription_key={my_subscription_key}&version=1.6.1' URL could not be accessed: HTTP/2 403
```

When I try to access the url in the browser, it downloads the zip package without any problems, when I try to `cURL` it on my remote server I get a `403 forbidden` response?

Anyone else seeing the same?
This happens on multiple projects at the moment.

Thanks!

**Comment by szepeviktor** [comment]
Is it the right URL?
If *yes* WPML's server responds with a clear HTTP/403.
Maybe they do not like `cURL`'s user agent string.
... or they expect a referrer.

**Comment by Twansparant** [comment]
Yes it's the right url, because it downloads the correct zip in a browser and it used to work perfectly for the last years.
I haven't changed anything and it suddenly stopped working...
Maybe they have changed something on their end?

**Comment by szepeviktor** [comment]
>  Maybe they have changed something on their end?

Try adding these to the HTTP request in PHP

>  Maybe they do not like cURL's user agent string.
> ... or they expect a referrer.

https://github.com/ffraenz/private-composer-installer/blob/main/src/RemoteFilesystem.php#L53
https://www.php.net/manual/en/context.http.php

**Comment by ffraenz** [comment]
I guess `{my_user_id}` and `{my_subscription_key}` are supposed to be placeholders? The correct syntax to specify placeholders is `{%ENV_VARIABLE_NAME}` (with a `%` before the variable name). When no placeholders are recognised this plugin ignores the package and leaves the dist URL unchanged. Does this help?

**Comment by Twansparant** [comment]
Thanks for your reply!
Yes they are placeholders but I omitted them here for security, in my composer.json they are correct:

```
    {
      "type": "package",
      "package": {
        "name": "wpml/gravityforms-multilingual",
        "version": "1.6.1",
        "type": "wordpress-plugin",
        "dist": {
          "type": "zip",
          "url": "https://wpml.org/?download=8882&user_id={%WPML_USER_ID}&subscription_key={%WPML_SUB_KEY}&version={%version}"
        },
        "require": {
          "composer/installers": "^1.9",
          "ffraenz/private-composer-installer": "^5.0"
        }
      }
    },
```
But this all used to work fine, my `composer.json` hasn't been changed nor has my remote server?

**Comment by leup** [comment]
Same here. Error message says the downloaded file is not a zip file.
Tried to download the file with cUrl and it did not work too.

**Comment by wcoppens** [comment]
Yep, they added some nasty checks on the UserAgent. Kinda sad they try to work against us.

**Comment by zenopopovici** [comment]
I've spent an hour with Support on this issue, it seems that in the docker machine somehow my API key with has special characters in it is not properly escaped (don't ask me why) ... I have dollar signs in my API key. The new keys are only alphanumeric. 

I think they will reset my key and this will solve the issue.

To debug your composer install try `composer instal -vvv` and check if the URL is correct. 

On my machine it works perfectly. When ran in Docker, it just cuts part of the key. I tried encoding it in several ways, it just won't work...

So check again if this is the case in your configuration.
